dds_cli/user.py

```diff
@@ -280,56 +280,60 @@ def check_token_file_permissions(self):
                 message=f"Token file permissions are not properly set, (got {permissions_readable} instead of required '-rw-------'). Please remove {self.token_file} and rerun the command."
             )
 
-    def token_expired(self, token):
+    def token_expired(self, token, log):
         """Check if the token has expired or is about to expire soon based on the UTC time.
 
         :param token: The DDS token that is obtained after successful basic and two-factor authentication.
             Token is already obtained before coming here, so not expected to be None.
 
         Returns True if the token has expired, False otherwise.
         """
+        current_utc_time = datetime.datetime.utcnow()
+
         expiration_time = self.__token_dates(token=token)
         time_to_expire = expiration_time - datetime.datetime.utcnow()
 
-        if expiration_time <= datetime.datetime.utcnow():
+        self.token_report(current_utc_time, expiration_time, log)
+
+        if expiration_time <= current_utc_time:
             LOG.debug("Token has expired. Now deleting it and fetching new token.")
             self.delete_token()
             return True
-        elif time_to_expire < dds_cli.TOKEN_EXPIRATION_WARNING_THRESHOLD:
-            LOG.warning(
-                f"Saved token will expire in {readable_timedelta(time_to_expire)}, "
-                f"please consider renewing the session using the 'dds auth login' command."
-            )
 
         return False
 
-    def token_report(self, token):
+    def token_report(self, current_utc_time, expiration_time, log):
```

Comment:
You've removed the `token` argument here, so running e.g. `dds auth info` fails with `TypeError: TokenFile.token_report() got an unexpected keyword argument 'token'`

Reply:
I have updated the `token_report` function.